src/rgw/rgw_op.cc

```diff
@@ -5280,28 +5280,13 @@ void RGWRestoreObj::execute(optional_yield y)
     return;
   }
   rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_MANIFEST);
-  if (attr_iter != attrs.end()) {
-    RGWObjManifest m;
-    decode(m, attr_iter->second);
-    RGWObjTier tier_config;
-    m.get_tier_config(&tier_config);
-    if (m.get_tier_type() == "cloud-s3") {
-      ldpp_dout(this, 20) << "execute: expiry days" << expiry_days <<dendl;
-      op_ret = handle_cloudtier_obj(s, this, driver, attrs, false, expiry_days, true, y);
-      if (op_ret < 0) {
-        ldpp_dout(this, 4) << "Cannot get cloud tiered object: " << *s->object
-        <<". Failing with " << op_ret << dendl;
-        if (op_ret == -ERR_INVALID_OBJECT_STATE) {
-          s->err.message = "This object was transitioned to cloud-s3";
-        }
-      }
-    } else {
-      ldpp_dout(this, 20) << "not cloud tier object erroring" << dendl;
-      op_ret = -ERR_INVALID_OBJECT_STATE;
+  op_ret = handle_cloudtier_obj(s, this, driver, attrs, false, expiry_days, true, y);
+  if (op_ret < 0) {
+    ldpp_dout(this, 4) << "Cannot get cloud tiered object: " << *s->object
+    <<". Failing with " << op_ret << dendl;
+    if (op_ret == -ERR_INVALID_OBJECT_STATE) {
+      s->err.message = "This object was transitioned to cloud-s3";
```

Comment:
error message here need to modified as well

Reply:
`s->err.message`  already seems to be set in handle_cloudtier_obj